sycl/include/CL/sycl/id.hpp

```diff
@@ -109,10 +109,6 @@ template <int dimensions = 1> class id : public detail::array<dimensions> {
 
 // OP is: ==, !=
 #ifndef __SYCL_DISABLE_ID_TO_INT_CONV__
-  using detail::array<dimensions>::operator==;
```

Comment:
clang in C++20 mode complains about ambiguity to the operator== when it's used from the base class, likely because of required conversions to the base class. So, defined `operator==` in this class.